vortex-file/src/read/mask.rs

```diff
@@ -213,13 +211,8 @@ impl RowMask {
             return Ok(Some(sliced.clone()));
         }
 
-        if (true_count as f64 / sliced.len() as f64) < PREFER_TAKE_TO_FILTER_DENSITY {
-            let indices = self.to_indices_array()?;
-            take(sliced, indices, TakeOptions::default()).map(Some)
-        } else {
-            let mask = FilterMask::try_from(self.bitmask.clone())?;
-            filter(sliced, mask).map(Some)
-        }
+        let mask = FilterMask::try_from(self.bitmask.clone())?;
+        filter(sliced, mask).map(Some)
     }
```

Comment:
Where does this logic live now?

Reply:
What logic?

Reply:
The one highlighted by this comment, it's not referring to single line

Reply:
<img width="929" alt="Screenshot 2024-12-01 at 09 21 54" src="https://github.com/user-attachments/assets/60074c4e-02a0-4336-b2fb-1f87f8a51f7f">

I don't know if that's working, I can't see what you highlighted.

In any case, the selectivity of a filter mask is now handled internally by all filter implementations. Most will iter_indices anyway, so has equal to or better performance than any equivalent take implementations.

Reply:
Github is being really helpful here. Since the underlying mask is a bitmask then producing indices out of it and taking is the same/slightly slower than iterating the bitmask in the filter implementation.

Reply:
Yeah, anywhere we _know_ we have sorted unique indices, we should prefer to use filter.